packages/fxa-content-server/app/tests/spec/models/account.js

```diff
@@ -2177,238 +2173,155 @@ describe('models/account', function() {
+        sessionTokenId: 'session-1',
+        lastAccessTime: 100,
+        lastAccessTimeFormatted: 'a few seconds ago',
         name: 'alpha',
+        userAgent: 'Firefox 50',
       });
+      sinon.spy(client, 'destroy');
 
-      sinon.stub(fxaClient, 'deviceDestroy').callsFake(function() {
+      sinon.stub(fxaClient, 'attachedClientDestroy').callsFake(function() {
         return Promise.resolve();
       });
 
-      return account.destroyDevice(device);
+      return account.destroyAttachedClient(client);
     });
 
-    it('tells the backend to destroy the device', function() {
+    it('tells the backend to destroy the client with correct ids', function() {
       assert.isTrue(
-        fxaClient.deviceDestroy.calledWith(SESSION_TOKEN, 'device-1')
+        fxaClient.attachedClientDestroy.calledWith(SESSION_TOKEN, {
+          clientId: null,
+          deviceId: 'device-1',
+          refreshTokenId: null,
+          sessionTokenId: 'session-1',
+        })
       );
+      assert.isTrue(client.destroy.calledOnce);
     });
   });
 
-  describe('destroyOAuthApp', function() {
-    var device;
+  describe('destroyAttachedClient for an oauth app', function() {
+    var client;
 
     beforeEach(function() {
-      account.set('accessToken', ACCESS_TOKEN);
+      account.set('sessionToken', SESSION_TOKEN);
 
-      device = new OAuthApp({
-        id: 'oauth-1',
+      client = new AttachedClient({
+        refreshTokenId: 'refresh-1',
+        clientId: 'client-1',
+        lastAccessTime: 100,
+        lastAccessTimeFormatted: 'a few seconds ago',
         name: 'alpha',
+        userAgent: 'Firefox 50',
       });
-    });
+      sinon.spy(client, 'destroy');
 
-    it('tells the backend to destroy the device', function() {
-      sinon.stub(account._oAuthClient, 'destroyOAuthApp').callsFake(function() {
+      sinon.stub(fxaClient, 'attachedClientDestroy').callsFake(function() {
         return Promise.resolve();
       });
 
-      return account.destroyOAuthApp(device).then(() => {
-        assert.isTrue(
-          account._oAuthClient.destroyOAuthApp.calledWith(
-            ACCESS_TOKEN,
-            'oauth-1'
-          )
-        );
-      });
+      return account.destroyAttachedClient(client);
     });
 
-    it('will retry if token is expired', function() {
-      sinon.stub(account, '_fetchProfileOAuthToken').callsFake(function() {
-        return Promise.resolve();
-      });
-      sinon.stub(account._oAuthClient, 'destroyOAuthApp').callsFake(function() {
-        return Promise.reject(OAuthErrors.toError('UNAUTHORIZED'));
-      });
-
-      return account.destroyOAuthApp(device).then(assert.fail, () => {
-        assert.isTrue(account._fetchProfileOAuthToken.called, 'fetch called');
-        assert.isTrue(
-          account._oAuthClient.destroyOAuthApp.calledTwice,
-          'called twice'
-        );
-      });
+    it('tells the backend to destroy the client with correct ids', function() {
```

Comment:
Should there be a test about destroying a refresh token w/o an access token (maybe they all expired?), or is that implied here?